scripts/react-compiler-compliance-check.ts

```diff
@@ -261,15 +262,163 @@ function createFilesGlob(filesToCheck?: string[]): string | undefined {
     return `**/+(${filesToCheck.join('|')})`;
 }
 
-function printFailureSummary({success, failures}: CompilerResults): void {
-    // const failedFileNames = getDistinctFileNames(Array.from(failures.values()), (f) => f.file, fileToCheck);
+/**
+ * Filters compiler results to only include failures for lines that were changed in the git diff.
+ * This helps focus on new issues introduced by the current changes rather than pre-existing issues.
+ *
+ * @param results - The compiler results to filter
+ * @param diffFilteringCommits - The commit range to diff (from and to)
+ * @returns Filtered compiler results containing only failures in changed lines
+ */
+async function filterResultsByDiff(
+    results: CompilerResults,
+    diffFilteringCommits: DiffFilteringCommits,
+    {shouldPrintSuccesses, shouldPrintSuppressedErrors}: PrintResultsOptions,
+): Promise<CompilerResults> {
+    // Check for uncommitted changes and warn if any exist
+    if (await Git.hasUncommittedChanges()) {
+        logWarn('Warning: You have uncommitted changes. The diff results may not accurately reflect your current working directory.');
+    }
+
+    logInfo(`Filtering results by diff between ${diffFilteringCommits.from} and ${diffFilteringCommits.to}...`);
+
+    // Get the diff between the two commits
+    const diffResult: DiffResult = Git.diff(diffFilteringCommits.from, diffFilteringCommits.to);
```

Comment:
Type can be inferred

```suggestion
    const diffResult = Git.diff(diffFilteringCommits.from, diffFilteringCommits.to);
```